connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/agent/DefaultAiAgentRequestHandler.java

```diff
@@ -0,0 +1,158 @@
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.agenticai.aiagent.agent;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.input.Prompt;
+import dev.langchain4j.model.input.PromptTemplate;
+import io.camunda.connector.agenticai.aiagent.memory.AgentContextChatMemoryStore;
+import io.camunda.connector.agenticai.aiagent.model.AgentContext;
+import io.camunda.connector.agenticai.aiagent.model.AgentMetrics;
+import io.camunda.connector.agenticai.aiagent.model.AgentResponse;
+import io.camunda.connector.agenticai.aiagent.model.AgentState;
+import io.camunda.connector.agenticai.aiagent.model.request.AgentRequest;
+import io.camunda.connector.agenticai.aiagent.model.request.AgentRequest.AgentRequestData.PromptConfiguration;
+import io.camunda.connector.agenticai.aiagent.provider.ChatModelFactory;
+import io.camunda.connector.agenticai.aiagent.tools.ToolCallingHandler;
+import io.camunda.connector.api.error.ConnectorException;
+import io.camunda.connector.api.outbound.OutboundConnectorContext;
+import java.util.Optional;
+
+public class DefaultAiAgentRequestHandler implements AiAgentRequestHandler {
+
+  private static final int DEFAULT_MAX_MODEL_CALLS = 10;
+  private static final int DEFAULT_MAX_MEMORY_MESSAGES = 20;
+
+  private final ObjectMapper objectMapper;
+  private final ChatModelFactory chatModelFactory;
+  private final ToolCallingHandler toolCallingHandler;
+
+  public DefaultAiAgentRequestHandler(
+      ObjectMapper objectMapper,
+      ChatModelFactory chatModelFactory,
+      ToolCallingHandler toolCallingHandler) {
+    this.objectMapper = objectMapper;
+    this.chatModelFactory = chatModelFactory;
+    this.toolCallingHandler = toolCallingHandler;
+  }
+
+  @Override
+  public AgentResponse handleRequest(OutboundConnectorContext context, AgentRequest request) {
+    final AgentRequest.AgentRequestData requestData = request.data();
+    final AgentContext agentContext =
+        Optional.ofNullable(requestData.context()).orElseGet(AgentContext::empty);
+
+    // initialize configured model
+    final ChatLanguageModel chatModel = chatModelFactory.createChatModel(request);
+
+    // set up memory and load from context if available
+    final AgentContextChatMemoryStore chatMemoryStore =
+        new AgentContextChatMemoryStore(objectMapper);
+    chatMemoryStore.loadFromAgentContext(agentContext);
+
+    final ChatMemory chatMemory =
+        MessageWindowChatMemory.builder()
```

Comment:
Keeps messages in chat memory according to the configured message window size. When it hits the limits if will first start to evict tool call requests + results and then continue evicting the oldest messages.

If we want to keep the whole conversation history (e.g. for RAG) we need to make sure to also keep track of that (will be a follow-up).